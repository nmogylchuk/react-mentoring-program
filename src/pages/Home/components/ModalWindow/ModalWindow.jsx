import React from 'react';
import {createPortal} from 'react-dom';
import CloseButton from 'pages/shared/CloseButton/CloseButton';
import 'pages/Home/components/ModalWindow/ModalWindow.scss';

const ModalWindow = (props) => {

    const domEl = document.getElementById("modal");
    if (!domEl) return null;

    return createPortal (
        <>
            <div className="modal-window">
                <CloseButton onClick={props.onClick}/>
                {props.children}
            </div>
            <div className="modal-window__background"></div>
        </>,
        domEl
    );
};

export default ModalWindow;
