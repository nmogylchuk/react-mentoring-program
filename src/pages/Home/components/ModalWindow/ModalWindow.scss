@import 'src/assets/styles/variables';
@import 'src/assets/styles/mixins';

.modal-window {
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 50%;
  padding: 30px;
  z-index: 4;
  background-color: $dark-grey;
  border: 1px solid $grey;

  &__background {
    @include background();
  }
}
