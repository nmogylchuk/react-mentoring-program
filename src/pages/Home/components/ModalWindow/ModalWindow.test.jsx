import React from 'react';
import rerender from 'react-test-renderer';
import ModalWindow from "./ModalWindow";

describe('when Modal Window', () => {


    test('then snapshot created', () => {
        const modalWindowComponent = rerender.create(<ModalWindow/>);
        const tree = modalWindowComponent.toJSON();
        expect(tree).toMatchSnapshot();
    });
});