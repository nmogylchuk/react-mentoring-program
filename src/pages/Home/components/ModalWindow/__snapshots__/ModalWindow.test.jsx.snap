// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when Modal Window then snapshot created 1`] = `null`;
